### PR TITLE
[VL] Simplify the usage of debug print by providing a macro DEBUG_OUT

### PR DESCRIPTION
**Pull request by zuochunwei**
## What changes were proposed in this pull request?

I designed a fake OStream class `FakeCout` to support the writing of `COUT << sth`, the empty implementation of `FakeCout::operator<<(T)` will be optimized out when the compiler's optimization switch is open. so don't worry about the unnecessary CPU time cost.

(Please fill in changes proposed in this fix)

(Fixes: \#ISSUE-ID)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: cpp/core/utils/Cout.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  }
+};
+
+extern FakeCout fakeCout;
+
+#define COUT fakeCout
+
+#endif
+
+} // namespace gluten
```
Comment (zhztheplayer) [inline cpp/core/utils/Cout.h:51]: Please add a EoF line. Thanks!

File: cpp/core/utils/Cout.h [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * limitations under the License.
+ */
+
+#pragma once
+
+namespace gluten {
+
+#ifdef GLUTEN_PRINT_DEBUG
+
+#define COUT std::cout
```
Comment (zhztheplayer) [inline cpp/core/utils/Cout.h:24]: `COUT` sounds like a regular print? Should it be named `DOUT` or something like that?

**Review by zhztheplayer — COMMENTED** [review]
Thanks!

File: cpp/core/utils/Cout.h (hunk 1 above)
Comment (zuochunwei) [reply cpp/core/utils/Cout.h:51]: OK

File: cpp/core/utils/Cout.h (hunk 2 above)
Comment (zuochunwei) [reply cpp/core/utils/Cout.h:24]: how about DEBUG_OUT

**Review by zhztheplayer — APPROVED** [review]
Thanks!

**Pull request merged**